src/frontend/js/data/useStaticFilters/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { useRef, useState } from 'react';
+
+import { FilterDefinition, StaticFilterDefinitions } from 'types/filters';
+import { Nullable } from 'utils/types';
+import { useAsyncEffect } from 'utils/useAsyncEffect';
+
+// Our search and autosuggestion pipeline operated based on filter definitions. Obviously, we can't filters
+// courses by courses, but we still need a filter-definition-like config to run courses autocompletion.
+const coursesConfig: FilterDefinition = {
+  base_path: null,
+  human_name: 'Courses',
+  is_autocompletable: true,
+  is_searchable: true,
+  name: 'courses',
+  position: 99,
+};
+
+export const useStaticFilters = (includeCoursesConfig = false) => {
```

Comment:
Hoe to use this filter is not obvious for me. Maybe specifying the return value can help to use it ?

Reply:
done, let me know if it's clearer now

Reply:
Not really, `useStaticFilters` does not return a `Promise`, it returns a function and this function returns a `Promise`. This is important to specify it to understand how to use it, without an example or reading the function it's not obvious.

Reply:
Oh right, my bad! Fixing this right now 😅